code/GetWayPoint.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Oct 30 14:36:59 2023
+
+@author: genie_god
+"""
+
+import configparser
+import time
+
+import pandas as pd
+import requests
+import tqdm
+
+
+def get_route_data(coordinates, access_token):
+    """
+    Fetch travel time list using Mapbox Direction API.
+    :param coordinates: List of coordinates [longitude, latitude]
+    :param access_token: Your Mapbox Access Token
+    :return: JSON response from Mapbox API
+    """
+    # Convert list of coordinates to string format
+    coordinates_str = ";".join([f"{lon},{lat}" for lon, lat in coordinates])
+
+    # Endpoint URL (assuming driving mode here,
+    # but can be changed to walking, cycling, etc.)
+    url_root = "https://api.mapbox.com/directions/v5/mapbox/driving"
+    url = f"{url_root}/{coordinates_str}?"
+
+    # Parameters
+    params = {
+        "access_token": access_token,
+        # "annotation": "duration",
+        "steps": "true",
+        "waypoints_per_route": "true",
+    }
+
+    # Make the API call
+    response = requests.get(url, params=params)
+    # Return the JSON response
+    return response.json()
+
+
+def add_coordinator(df):
+    df["Coordinates"] = df[["Longitude", "Latitude"]].values.tolist()
+    df.drop(columns=["Longitude", "Latitude"], inplace=True)
+    return df
+
+
+def initialize_data():
+    """
+    Initialize the data from the CSV file
+    and the Mapbox token from the config file.
+
+    :return: Dataframe and Mapbox token
+    """
+
+    # Initialize the parser
+    config = configparser.ConfigParser()
+    # Read the config file
+    config.read("config.ini")
+    mapbox_token = config["mapbox"]["token"]
+
+    # Take file name from terminal
+    file_name = "../data/route1.csv"
+    df1 = pd.read_csv(file_name)
+    file_name = "../data/route2.csv"
```

Comment:
Don't hardcode the filenames here, even if you think the filenames will have consistent naming format

Reply:
updated this part as dir will be input from config file